nuxeo-core/nuxeo-core-mongodb/src/test/java/org/nuxeo/ecm/core/mongodb/TestMongoDBIndices.java

```diff
@@ -0,0 +1,128 @@
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.runtime.mongodb.MongoDBFeature;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+
+/**
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features({ MongoDBFeature.class, CoreFeature.class })
+@Deploy("org.nuxeo.ecm.core.mongodb.test:OSGI-INF/mongodb-disable-renaming-listener-test.xml")
+@Deploy("org.nuxeo.ecm.core.mongodb.test:OSGI-INF/mongodb-enable-unique-indices-contrib.xml")
+public class TestMongoDBIndices {
+
+    protected static final String DOCUMENT_NAME = "anyFile";
+
+    @Inject
+    protected CoreSession coreSession;
+
+    protected DocumentModel domain;
+
+    @Before
+    public void before() {
+        domain = coreSession.createDocumentModel("/", "Folder", "Folder");
+        domain = coreSession.createDocument(domain);
+        addPermissions(domain, "james");
+        addPermissions(domain, "bob");
+        coreSession.save();
+    }
+
+    @Test
+    public void shouldFailsWhenCreateExistingChildNameDocument() {
+        try (CloseableCoreSession bobSession = CoreInstance.openCoreSession(coreSession.getRepositoryName(), "bob")) {
+            createDocument(bobSession, domain, DOCUMENT_NAME);
+        }
+
+        try (CloseableCoreSession jamesSession = CoreInstance.openCoreSession(coreSession.getRepositoryName(),
+                "james")) {
+            createDocument(jamesSession, domain, DOCUMENT_NAME);
+            fail("should throw a ConcurrentUpdateException");
+        } catch (ConcurrentUpdateException cue) {
+            assertEquals(SC_CONFLICT, cue.getStatusCode());
+            assertTrue(
+                    cue.getMessage()
+                       .contains(String.format(
+                               "E11000 duplicate key error collection: %s.test index: ecm:parentId_1_ecm:name_1 dup key: { : \"%s\", : \"%s\" }",
+                               DEFAULT_MONGODB_DBNAME, domain.getId(), DOCUMENT_NAME)));
+        }
+    }
+
+    @Test
+    public void shouldFailsWhenMovingDocumentToExistingChildName() {
+        try (CloseableCoreSession bobSession = CoreInstance.openCoreSession(coreSession.getRepositoryName(), "bob")) {
+            createDocument(bobSession, domain, DOCUMENT_NAME);
+        }
+
+        try (CloseableCoreSession jamesSession = CoreInstance.openCoreSession(coreSession.getRepositoryName(),
+                "james")) {
+            DocumentModel documentModel = createDocument(jamesSession, domain, "jamesFileName");
+
+            // org.nuxeo.ecm.core.storage.dbs.DBSSession.move (Objects.equals(oldParentId, parentId)) checks the unique
+            // name before calling the mongodb update backend
+            documentModel = jamesSession.move(documentModel.getRef(), null, DOCUMENT_NAME);
+            jamesSession.saveDocument(documentModel);
+            jamesSession.save();
```

Comment:
Isn't needed no?

Reply:
yes i agree, i just kept it :).

Reply:
Why?

Reply:
it's was mainly to have the same scenario then the creation where i create the document (the update is done by move in this case) and then i save it and the session juste after.